### DISCUSSION TITLE
应该如何自定义help菜单呢

### DISCUSSION THREAD

**Discussion by ID114514** [Arondight/Adachi-BOT#295] [Q&A] [answered]
输入help后弹出的菜单列表太长了，想适当减掉一部分长度

**Comment by Arondight** [comment, marked as answer]
https://github.com/Arondight/Adachi-BOT/blob/75dff6c3b17e8fed46c747a826d273c893172a90/docs/%E5%BC%80%E5%8F%91%E6%8C%87%E5%BC%95.md#%E4%BB%A3%E7%A0%81%E7%A4%BA%E4%BE%8B

```
如果你希望插件中 functions 的某个功能生效但是不显示在帮助信息中，把该功能的 name 置空即可。
```

> 已作废，现在增加了字段 `show` 来做这件事情

**Reply by Arondight** [reply]
我觉得或许我需要做一个字段来显式指定某个插件或者功能是否要显示在帮助信息里，简单的置空 name 容易造成问题和误解

**Reply by Arondight** [reply]
#296

**Reply by ID114514** [reply]
还有就是如何 添加 修改 触发关键词呢，直接在此文本中添加 貌似不会生效

**Reply by Arondight** [reply]
> 
> 
> 还有就是如何 添加 修改 触发关键词呢，直接在此文本中添加 貌似不会生效

需要修改 regex 和 entrance 字段， name 、 usage 、 description 没有实际作用

**Reply by ID114514** [reply]
好的了解了，不过还有一个疑问，如何取消关键词后面的空格，例如 uid 111111111 中间需要添加空格bot才会进行查询

**Reply by Arondight** [reply]
> 
> 
> 好的了解了，不过还有一个疑问，如何取消关键词后面的空格，例如 uid 111111111 中间需要添加空格bot才会进行查询

取消不了，需要该插件，插件中使用空白字符来分割关键词的，为的是去除硬编码

**Reply by mark9804** [reply]
> 好的了解了，不过还有一个疑问，如何取消关键词后面的空格，例如 uid 111111111 中间需要添加空格bot才会进行查询

**如果不知道相关插件是怎么运作的，并且没有JS相关知识的话，最好不要改动，因为这样有可能导致相关插件不能正常工作**。
<details>
<summary>如果执意需要改动：</summary>
如果执意需要改动：

首先，把`config_defaults/command.yml`复制一份到`config`目录下面，以免之后拉取新代码时覆盖你的修改

然后，编辑`config/command.yml`，找到这一行：
https://github.com/Arondight/Adachi-BOT/blob/75dff6c3b17e8fed46c747a826d273c893172a90/config_defaults/command.yml#L342

改成：
`    - ^uid\s*\d+\s*$`（注意英文破折号前面有四个空格）
保存。

然后，在Adachi-BOT的根目录下，运行：
```shell
npm run restart
```

尝试能不能正常使用。**如果不能正常使用该功能并且不知道怎么修改相关插件代码使其能够正常执行，请把`config/command.yml`改回来并且重新执行`npm run restart`，然后就不要去碰了。之后就使用原始提供的格式查询。**
</details>

**Reply by Arondight** [reply]
拉一下代码，我把新字段加上了

**Reply by Arondight** [reply]
@Mark9804 你的方法只修改了入口，实际上插件内还有一个正则用来切分关键字，也需要修改

**Reply by mark9804** [reply]
诶，我看`id.js`是用`/\d+/g`来切分的，可能我找错函数了

确实需要修改插件，我建议不要尝试，保持现在的`uid xxxxxxxxx`的格式查询
